src/charm.py

```diff
@@ -292,6 +306,10 @@ def _get_auth_devices_keys_from_db(self):
             logger.error(f"Failed to fetch auth devices keys from '{database_url}': {e}")
             return None
 
+    def _on_invalid_probe(self, event) -> None:
+        """Check from invalid probes events and set the charm to Blocked if so."""
+        self.unit.status = BlockedStatus("Invalid probe provided")
```

Comment:
Here you could be overwriting (losing) a previously set status.
See https://github.com/canonical/operator/pull/954 for an alternative, centralized, approach.